core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -330,4 +303,103 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
         ? droppedFromStore
         : droppedFromCatalog;
   }
+
+  private boolean importSchema(NameIdentifier identifier) {
+    EntityCombinedSchema combinedSchema = loadCombinedSchema(identifier);
+    if (combinedSchema.imported()) {
+      return false;
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(combinedSchema.schemaProperties());
+    long uid;
+    if (stringId != null) {
+      // If the entity in the store doesn't match the external system, we use the data
+      // of external system to correct it.
+      uid = stringId.id();
+    } else {
+      // If store doesn't exist entity, we sync the entity from the external system.
+      uid = idGenerator.nextId();
+    }
+
+    SchemaEntity schemaEntity =
+        SchemaEntity.builder()
+            .withId(uid)
+            .withName(identifier.name())
+            .withNamespace(identifier.namespace())
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(combinedSchema.auditInfo().creator())
+                    .withCreateTime(combinedSchema.auditInfo().createTime())
+                    .withLastModifier(combinedSchema.auditInfo().lastModifier())
+                    .withLastModifiedTime(combinedSchema.auditInfo().lastModifiedTime())
+                    .build())
+            .build();
+    try {
+      store.put(schemaEntity, true);
+    } catch (Exception e) {
+      LOG.error(FormattedErrorMessages.STORE_OP_FAILURE, "put", identifier, e);
+      throw new RuntimeException("Fail to access underlying storage");
```

Comment:
I think you need to pass the exception `e` to `RuntimeException`.

Reply:
If I pass the `e`, users will find a situation.
```
When they load a load table, it may throw a write exception.
```